[cmdpal] Add fallback item for system command.

moooyo opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request
Introduces a new fallback command feature for system commands. 
![image](https://github.com/user-attachments/assets/d20bbb9f-b56f-4a3c-86ad-31648d96a46d)


Key changes include:
- `EmptyRecycleBinCommand` and `EmptyRecycleBinConfirmation` classes moved to the `Microsoft.CmdPal.Ext.System` namespace. 
- Introduction of the `FallbackSystemCommandItem` class to manage fallback commands based on user settings.
- Updates to `SystemCommandExtensionProvider` to include a new fallback command item and its retrieval method.
- Addition of a new localized string for `Microsoft_plugin_ext_fallback_display_title` in `Resources.Designer.cs`.
- Complete removal of the `SystemCommandsCache` class and its associated logic.

By default we include all sys command in the fallback command item. Maybe we need a configuration to make user control it?

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [x] **Closes:** #38462
- [ ] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [x] **Tests:** Added/updated and all pass
- [ ] **Localization:** All end user facing strings can be localized
- [ ] **Dev docs:** Added/updated
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: #xxx

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments

<!-- Describe how you validated the behavior. Add automated tests wherever possible, but list manual validation steps taken as well -->
## Validation Steps Performed

moooyo commented on src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.System/FallbackSystemCommandItem.cs:
```diff
@@ -0,0 +1,65 @@
+        {
+            var title = command.Title;
+            var subTitle = command.Subtitle;
+            var titleScore = StringMatcher.FuzzySearch(query, title).Score;
```
Maybe we just need a exactly match? Seems calculate score again and again will cost more resource.

htcfreek commented:
we should skip network results in fallback commands. No need and it might cause delays on network connection errors.

---

can we do the fallback command for time, date and now results of time plugin too?

htcfreek commented:
@moooyo 
Did your change break the network results on system plugin page? They were extra requested in the removed cache.

moooyo commented:
> we should skip network results in fallback commands.

Agree, they were separated by two function (GetSystemCommands and GetNetworkConnectionResults). Will not exist in fallback item.

moooyo commented:
> can we do the fallback command for the results time, date and now of time plugin too? (separate pr)

Sure, we need to create another task to track it.

moooyo commented:
> @moooyo Did your change break the network results on system plugin page? They were extra requested in the removed cache.

No... actually, they were not use cache. I implemented the cache class but never use it. So, I removed it in this PR.

moooyo commented:
> > @moooyo Did your change break the network results on system plugin page? They were extra requested in the removed cache.
> 
> No... actually, they were not use cache. I implemented the cache class but never use it. So, I removed it in this PR.

But there shouldn't be any extra cost. They should be cached in the viewmodel, right?

moooyo commented:
> can we do the fallback command for the results time, date and now of time plugin too? (separate pr)

Created https://github.com/microsoft/PowerToys/issues/38890

htcfreek commented:
> > > @moooyo Did your change break the network results on system plugin page? They were extra requested in the removed cache.
> > 
> > No... actually, they were not use cache. I implemented the cache class but never use it. So, I removed it in this PR.
> 
> But there shouldn't be any extra cost. They should be cached in the viewmodel, right?

Ooooh. Don't know that.

moooyo commented:
lol, you know now.
